vms/platformvm/txs/mempool/mempool.go

```diff
@@ -121,6 +124,15 @@ func New(
 	}, nil
 }
 
+func (m *mempool) Iterate(f func(tx *txs.Tx) bool) {
+	itr := m.unissuedTxs.NewIterator()
```

Comment:
q: how about concurrency here? Is mempool locking for iteration up to the client of `Iterate`?

Reply:
The iterator internally has a reference to the lock to the map

Reply:
Nvm we have to lock the entire mempool up for this iteration because the iterator assumes that there's no concurrent updates to this map